src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -1,38 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-/**
- * Parameterize this service and add its implementation in a separate class.
- */
-public interface MachineService {
+public interface MachineService<T extends Machine> {
+    List<Machine> getAll(Class<? extends Machine> type);
```

Comment:
Consider parameterizing the return type of the `getAll` method to match the class type parameter `T`. This would ensure type safety and make the method signature more consistent with the generic class definition.